LiteCore/Query/QueryParser+VectorSearch.cc

```diff
@@ -37,10 +37,14 @@ namespace litecore {
         slice metricName;
         if ( auto metricVal = params[2] )
             metricName = requiredString(metricVal, "3rd argument (metric) to APPROX_VECTOR_DISTANCE");
+            // The following requirement does not apply to the lazy index. At this place, we don't know
+            // if uses lazy index.
+#    if 0
         require(expr->type() == kArray,
                 "first argument to APPROX_VECTOR_DISTANCE must evaluate to a vector; did you pass the index name %s "
                 "instead?",
                 exprJSON.c_str());
+#    endif
```

Comment:
I think we just remove this. There is no need to do `if 0`. We could get the code back from git.

Reply:
I can do it